### DISCUSSION TITLE
[Feature Request]: make feather-icons WYSIWYG

### DISCUSSION THREAD

**Discussion by warpdesign** [GetPublii/Publii#1369] [Ideas]
### Feature Description

Adding a new feather-icon in (eg. `[fi=github]`)_WYSIWYG_ editor shows the shortcode, not what you see.

Editor:
![image](https://github.com/GetPublii/Publii/assets/199648/965e748c-6d1f-4f63-a360-d36afc6e9a54)

What you get:
![image](https://github.com/GetPublii/Publii/assets/199648/af5be432-6de7-4896-aa7e-76f00eafd0f8)

There should be an easy way to add a feather icon, and the icon should be visible in the WYSIWYG editor.

**Comment by bobmitro** [comment]
While I think your idea is great, I regret to inform you that our plugin API does not currently support this kind of implementation. However, we are continually working to improve our API's capabilities, and we will consider integrating this feature as soon as possible. Please stay tuned for future updates.

**Comment by HowToMeetLadies** [comment]
Hello, I just have a little suggestion so I think it's okay if I post it here.

The description says the following:
> Please make sure to include the elements in this exact sequence: icon name, size, color (in HEX format), class, aria-hidden, and stroke-width. The color should be specified using HEX format. This format is necessary for the plugin to correctly interpret and display the icons as intended.

I suggest taking the parameter evaluation out of the regex and moving it into the function.

```
return c.replace(/\[fi=([^ ]+)((?: [a-z-]+=[^=\]]+)*)\]/gi, (match, iconName, attr) => {
      const opts = (attr || "")
        .trim().replace(/\s+/g, ' ')  // tidy
        .split(' ')                   // ["key=value"]
        .map(kv => kv.split('='))     // [[key, value]]
        .filter(                      // only valid
          arr => ["size", "color", "class", "aria-hidden", "stroke-width"].includes(arr[0]))
        .reduce(                      // { key: value }
          (dict, arr) => { dict[arr[0]] = arr[1]; return dict },
          { "size": "24px", "color": "currentColor", "class": "", "aria-hidden": "false", "stroke-width": "2" });
});
```
(It's an excerpt from my session today. However, it shows the evaluation of the shortcode parameters, in any order.)


---
A manual tinymce integration (config override) might follow, but I'm not promising it.

**Reply by HowToMeetLadies** [reply]
I wrote it as PoC, but it works so far: [fifi live shortcodes (hack)](https://p4p.pages.dev/plugin-fifi/).

WYSIWYG:
<img width="714" alt="fifi-wysiwyg" src="https://github.com/GetPublii/Publii/assets/755996/95adcf75-d027-470c-9864-2b735f53a236">

Source:
<img width="1030" alt="fifi-html" src="https://github.com/GetPublii/Publii/assets/755996/1b83f2af-579b-494c-8f38-268e392da6c1">
